File: sol/2750.cpp
struct Solution {
    static constexpr int m = 1e9 + 7;
    int numberOfGoodSubarraySplits(vector<int>& nums) {
        int res = 1, prev = -1;
        for (int i = 0; i < nums.size(); ++i){
            if (nums[i] != 1){
                continue;
            }
            if (prev != -1){
                res = (static_cast<long>(res) * (i - prev)) % this->m;
            }
            prev = i;
        }
        return prev == -1 ? 0 : res;
    }
};

File: sol/2750.py
class Solution:
    m = 10 ** 9 + 7
    def numberOfGoodSubarraySplits(self, nums: List[int]) -> int:
        res = 1
        prev = -1
        for i, x in enumerate(nums):
            if x != 1:
                continue
            if prev != -1:
                res = (res * (i - prev)) % self.m
            prev = i
        return res if prev != -1 else 0
